backend/typescript/services/implementations/camperService.ts

```diff
@@ -127,6 +127,64 @@ class CamperService implements ICamperService {
 
     return camperDtos;
   }
+
+  /* eslint-disable class-methods-use-this */
+  async updateCamperById(
+    camperId: string,
+    camper: UpdateCamperDTO,
+  ): Promise<CamperDTO> {
+    let oldCamper: Camper | null;
+
+    try {
+      oldCamper = await MgCamper.findById(camperId);
+
+      if (oldCamper) {
+        const newCamp: Camp | null = await MgCamp.findById(camper.camp);
+        const oldCamp: Camp | null = await MgCamp.findById(oldCamper.camp);
+
+        if (
+          newCamp &&
+          oldCamp &&
+          newCamp.baseCamp.toString() !== oldCamp.baseCamp.toString()
+        ) {
+          throw new Error(
+            `Error: can only change sessions between the same camp`,
+          );
+        }
+      }
+
+      // must explicitly specify runValidators when updating through findByIdAndUpdate
+      oldCamper = await MgCamper.findByIdAndUpdate(
+        camperId,
+        {
+          // camp: camper.camp,
+          formResponses: camper.formResponses,
+          dropOffType: camper.dropOffType,
+          registrationDate: camper.registrationDate,
+          hasPaid: camper.hasPaid,
+          chargeId: camper.chargeId,
+        },
+        { runValidators: true, omitUndefined: true },
```

Comment:
you might have to remove omitUndefined - it became deprecated when we updated our mongoose version (do you get any deprecation warnings when start your server? :o). I believe findByIdAndUpdate should omit undefined fields by default now, would you be able to test this and check?